### PR TITLE
test: add Handlebars basic example

### PR DESCRIPTION
**Pull request by LeBenLeBen**
This is almost the default you get when running `fractal new`, I just added npm scripts. I plan on adding more components and pages to test more functionalities later and then duplicate it for Nunjucks and Twig adapter.

Relates to #457

### CODE REVIEW

File: packages/handlebars/package.json [hunk 1]
```diff
@@ -21,6 +21,9 @@
     "promised-handlebars": "^2.0.1"
   },
   "peerDependencies": {
-    "@frctl/fractal": "^1.1.0"
+    "@frctl/fractal": "^1.3.0"
```
Comment (mihkeleidast) [inline packages/handlebars/package.json:24]: should not update the peerDependency version since we don't change any functionality

**Comment by mihkeleidast** [comment]
And as discussed in Discord, let's add some actual tests with the example.

File: packages/handlebars/package.json (hunk 1 above)
Comment (LeBenLeBen) [reply packages/handlebars/package.json]: Reverted

File: examples/handlebars/docs/01-index.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+title: Fractal default example
+---
+
+This is your index page. You can edit its contents at `docs/01-index.hbs`
```
Comment (mihkeleidast) [inline examples/handlebars/docs/01-index.md:5]: the extension should be md here i think? is this a bug in our `new` command?

File: packages/handlebars/package.json [hunk 3]
```diff
@@ -22,5 +22,8 @@
   },
   "peerDependencies": {
     "@frctl/fractal": "^1.1.0"
+  },
+  "devDependencies": {
```
Comment (mihkeleidast) [inline packages/handlebars/package.json:26]: can you rebase this on master? i added the devDeps to this file in #616, but in a slightly different location - just so this won't start conflicting

Comment (LeBenLeBen) [reply packages/handlebars/package.json:29]: Done

File: examples/handlebars/docs/01-index.md (hunk 2 above)
Comment (LeBenLeBen) [reply examples/handlebars/docs/01-index.md]: Yes, was a bug, fixed it.

File: examples/handlebars/components/notes/notes.spec.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
+const fractal = require('../../fractal.config.js');
+
+describe('tree', () => {
+    beforeEach(async () => {
+        await fractal.load();
+    });
+
+    it('loads notes from config', async () => {
```
Comment (mihkeleidast) [inline examples/handlebars/components/notes/notes.spec.js:8]: we should also look if a variant with notes that's defined in the config gets the proper notes.

File: packages/fractal/views/cli/new/docs/index.hbs [hunk 5]
```diff
@@ -2,4 +2,4 @@
 title: {{ projectTitle }}
```
Comment (mihkeleidast) [inline packages/fractal/views/cli/new/docs/index.hbs:2]: this file itself should also be renamed to md extension

File: packages/handlebars/package.json (hunk 3 above)
Comment (mihkeleidast) [reply packages/handlebars/package.json:29]: looks like this is still doubled

File: examples/handlebars/package.json [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+    "test": "jest",
+    "test:dev": "npm run test -- --watch"
+  },
+  "dependencies": {
+    "@frctl/fractal": "^1.3.0",
+    "@frctl/handlebars": "^1.2.0",
+    "@frctl/mandelbrot": "^1.4.0"
+  },
+  "devDependencies": {
+    "jest": "^26.0.1",
```
Comment (Chapabu) [inline examples/handlebars/package.json:17]: I have a bit of an issue with this, but it's one that we may need to chat about in Slack.

We don't have many tests, but where we do - they're using Mocha/Chai/Sinon. The reason I'm not saying "Nope, we should change this" and "We should chat" instead, is that I'd REALLY like to migrate all our tests to Jest instead to reduce the amount of dependencies we need (as Jest literally covers everything we need).

Comment (mihkeleidast) [reply examples/handlebars/package.json:17]: I'm not against switching to Jest completely, but I'd wager that could be another issue/pr for another day. We don't exactly lose anything by having multiple test runners for a while. So if we could write this (migrating away from mocha/chai/sinon) up as an issue, I wouldn't block this PR.

**Comment by Chapabu** [comment]
Got conflicts in the lockfile. Do we need to even commit these?

**Comment by mihkeleidast** [comment]
> Got conflicts in the lockfile. Do we need to even commit these? They don't get published to NPM anyway, so they're only useful for contributors.

Yup, useful for contributors + for running tests predictably. No lockfile means that if a dependency breaks anything in a minor or patch, our tests will start failing. Of course, users will still get the breaking dependency versions, but the version diff will then be a good place to start looking for what went wrong.

File: examples/handlebars/package.json (hunk 6 above)
Comment (LeBenLeBen) [reply examples/handlebars/package.json:17]: Yeah I'm also in favor of going full Jest, but I'd like to go step by step on this one as Mihkel said.

**Comment by LeBenLeBen** [comment]
I should have time to tackle the remaining comments on Monday.

File: packages/handlebars/package.json (hunk 3 above)
Comment (LeBenLeBen) [reply packages/handlebars/package.json]: Should be ok now

File: examples/handlebars/components/notes/notes.spec.js (hunk 4 above)
Comment (LeBenLeBen) [reply examples/handlebars/components/notes/notes.spec.js:8]: Done

File: packages/fractal/views/cli/new/docs/index.hbs (hunk 5 above)
Comment (LeBenLeBen) [reply packages/fractal/views/cli/new/docs/index.hbs:2]: Done

File: examples/README.md [hunk 7]
```diff
@@ -0,0 +1,5 @@
+# Fractal examples
+
+This folder contains examples of Fractal instances using various adapters & settings.
+
+Although we use them for developing and testing Fractal, they are probably a great ressource for users as well.
```
Comment (mihkeleidast) [inline examples/README.md:5]:
```suggestion
Although we use them for developing and testing Fractal, they are probably a great resource for users as well.
```

File: examples/handlebars/README.md [hunk 8]
```diff
@@ -0,0 +1,33 @@
+# Fractal Handlebars example
+
+This is an example instance of Fractal using the default Handlebars adapter.
+
+## Setup
+
+[Follow these instructions](https://github.com/frctl/fractal#development) to setup the development environement, then run:
+
+```bash
+npm start
```
Comment (mihkeleidast) [inline examples/handlebars/README.md:10]: blabbed about this under the Mandelbrot PR, but we should somehow agree on a development process and figure out what the best way to get started is when getting the repo for the first time. Relevant point here is that before running the example instance, you need to build/start mandelbrot.

Comment (mihkeleidast) [reply examples/handlebars/README.md:10]: Cross-posting, since it's a bit relevant under both PRs:
> Found that running npx lerna run --scope=@frctl/handlebars-example --include-dependencies --parallel start for the example project will build mandelbrot (and any other dependencies that have a start script) too.
